File: hot_chocolate_small/dps/entities/Shipment.cs
using System;
using System.Collections.Generic;

using HotChocolate;

using x10.hotchoc.dps.Repositories;

namespace x10.hotchoc.dps.Entities {
  // Enums
  public enum TransportationModeEnum {
    Air,
    Ocean,
    Truck,
    Rail,
    UnknownTransportation,
    TruckIntl,
    WarehouseStorage,
  }

  public enum ShipmentPriorityEnum {
    Standard,
    High,
  }


  /// <summary>
  /// A portion of Core Shipment entity
  /// </summary>
  public class Shipment : Base {
    // Regular Attributes
    public int? Dbid { get; set; }
    [GraphQLNonNullType]
    public string? FlexId { get; set; }
    [GraphQLNonNullType]
    public string? Name { get; set; }
    public ShipmentPriorityEnum? Priority { get; set; }
    public TransportationModeEnum? TransportationMode { get; set; }
    [GraphQLNonNullType]
    public string? Status { get; set; }
    public DateTime? CargoReadyDate { get; set; }
    public DateTime? ActualDepartureDate { get; set; }
    public DateTime? ArrivalDate { get; set; }
    [GraphQLNonNullType]
    public bool IsLcl { get; set; }
    [GraphQLNonNullType]
    public bool IsLtl { get; set; }
    [GraphQLNonNullType]
    public string? Customs { get; set; }
    public DateTime? DueDate { get; set; }
    [GraphQLNonNullType]
    public string? DueDateTask { get; set; }

    // To String Representation
    [GraphQLNonNullType]
    public string? ToStringRepresentation {
      get { return "Shipment: " + DbidHotChoc; }
      set { /* Needed to make Hot Chocolate happy */ }
    }

    // Associations
    public CompanyEntity? Consignee { get; set; }
    public CompanyEntity? Shipper { get; set; }
    public Port? DeparturePort { get; set; }
    public Port? ArrivalPort { get; set; }

    public override void EnsureUniqueDbid() {
      base.EnsureUniqueDbid();
    }

    internal override void SetNonOwnedAssociations(IRepository repository) {
      base.SetNonOwnedAssociations(repository);

      int? consignee = IdUtils.FromRelayId(Consignee?.Id);
      Consignee = consignee == null ? null : repository.GetCompanyEntity(consignee.Value);

      int? shipper = IdUtils.FromRelayId(Shipper?.Id);
      Shipper = shipper == null ? null : repository.GetCompanyEntity(shipper.Value);

      int? departurePort = IdUtils.FromRelayId(DeparturePort?.Id);
      DeparturePort = departurePort == null ? null : repository.GetPort(departurePort.Value);

      int? arrivalPort = IdUtils.FromRelayId(ArrivalPort?.Id);
      ArrivalPort = arrivalPort == null ? null : repository.GetPort(arrivalPort.Value);
    }
  }
}

File: platforms/flow/react_small_generated/x10_generated/dps/entities/Shipment.js
// TEAM: compliance
// @flow

import { v4 as uuid } from 'uuid';

import { addError, type FormError } from 'react_lib/form/FormProvider';
import isBlank from 'react_lib/utils/isBlank';
import toEnum from 'react_lib/utils/toEnum';
import x10toString from 'react_lib/utils/x10toString';

import { type CompanyEntity } from 'dps/entities/CompanyEntity';
import { type Port } from 'dps/entities/Port';
import { type TransportationModeEnum } from 'dps/sharedEnums';


// Type Definition
export type Shipment = {
  +id: string,
  +dbid: ?number,
  +flexId: string,
  +name: string,
  +priority: ?ShipmentPriorityEnum,
  +transportationMode: ?TransportationModeEnum,
  +status: string,
  +cargoReadyDate: ?string,
  +actualDepartureDate: ?string,
  +arrivalDate: ?string,
  +isLcl: boolean,
  +isLtl: boolean,
  +customs: string,
  +dueDate: ?string,
  +dueDateTask: string,
  +consignee: ?CompanyEntity,
  +shipper: ?CompanyEntity,
  +departurePort: ?Port,
  +arrivalPort: ?Port,
};


// Enums
export const ShipmentPriorityEnumPairs = [
  {
    value: 'standard',
    label: 'Standard',
  },
  {
    value: 'high',
    label: 'High',
  },
];

export type ShipmentPriorityEnum = 'standard' | 'high';



// Derived Attribute Functions
export function shipmentUrl(shipment: ?{
  +dbid: ?number,
}): string {
  if (shipment == null) return '';
  const result = '/shipments/' + x10toString(shipment?.dbid);
  return result;
}

export function shipmentModeSubtext(shipment: ?{
  +transportationMode: ?TransportationModeEnum,
  +isLcl: boolean,
  +isLtl: boolean,
}): string {
  if (shipment == null) return '';
  const result = toEnum(shipment?.transportationMode) == "ocean" ? (shipment?.isLcl ? 'LCL' : 'FCL') : toEnum(shipment?.transportationMode) == "truck" ? (shipment?.isLtl ? 'LTL' : 'FTL') : '';
  return result;
}



// Create Default Function
export function createDefaultShipment(): Shipment {
  return {
    id: uuid(),
    dbid: null,
    flexId: '',
    name: '',
    // $FlowExpectedError Required field, but no default value
    priority: null,
    // $FlowExpectedError Required field, but no default value
    transportationMode: null,
    status: '',
    cargoReadyDate: null,
    actualDepartureDate: null,
    arrivalDate: null,
    isLcl: false,
    isLtl: false,
    customs: '',
    dueDate: null,
    dueDateTask: '',
    consignee: null,
    shipper: null,
    departurePort: null,
    arrivalPort: null,
  };
}


// Validations
export function shipmentCalculateErrors(shipment: Shipment, prefix?: string, inListIndex?: number): $ReadOnlyArray<FormError> {
  const errors = [];
  if (shipment == null ) return errors;

  if (isBlank(shipment.priority))
    addError(errors, prefix, 'Priority is required', ['priority'], inListIndex);
  if (isBlank(shipment.transportationMode))
    addError(errors, prefix, 'Transportation Mode is required', ['transportationMode'], inListIndex);

  return errors;
}
